pkg/build/controller/factory/factory.go

```diff
@@ -95,6 +96,10 @@ func (factory *BuildControllerFactory) Create() controller.RunnableController {
 			build := obj.(*buildapi.Build)
 			err := buildController.HandleBuild(build)
 			if err != nil {
+				// Set default Reason.
+				if build.Status.Reason == "" {
```

Comment:
It's a convention we follow in our codebase to not compare against empty strings.